lib/features/manager.js

```diff
@@ -0,0 +1,157 @@
+// @ts-check
+const {
+  loadSqlEquiv,
+  queryAsync,
+  queryValidatedSingleColumnOneRow,
+} = require('@prairielearn/postgres');
+const { z } = require('zod');
+
+const config = require('../config');
+
+const sql = loadSqlEquiv(__filename);
+
+const CONTEXT_HIERARCHY = ['institution_id', 'course_id', 'course_instance_id'];
+const DEFAULT_CONTEXT = {
+  institution_id: null,
+  course_id: null,
+  course_instance_id: null,
+  user_id: null,
+};
+
+/**
+ * Grants can have different types, which can be used to determine how they
+ * were applied. These are mostly meant to be consumed by administrators who
+ * need to be able to tell if a feature was enabled by default, set manually
+ * by another administrator, or enabled as part of a subscription plan.
+ *
+ * @enum {string}
+ */
```

Comment:
Thoughts on this? Are there any other types of grants we should account for?

Reply:
I'm not sure if 'grant' is the best word to use here, especially considering that most instructors using PL are probably primed to think about writing applications to the NSF for funding when they see the word "grant". Is there a different word we could use?

Reply:
Instructors never see table names, or implementation details in general, so that shouldn't matter. I'm open to suggestions if you have any.

Reply:
Just to pick a couple of semi-random examples, Hashicorp's RBAC Boundary product uses [grants](https://developer.hashicorp.com/boundary/docs/concepts/security/permissions/permission-grant-formats) for this concept. Kubernetes uses [RoleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) for it's RBAC model.

Reply:
This all sounds good to me. No further grant types that I can think of!